File: cpp/Solution2582.cpp
/*
submitted at Jul 06, 2024 15:07

Runtime 0 ms, Beats 100.00%
Memory 7.02 MB, Beats 71.18%
*/

class Solution {
public:
    int passThePillow(int n, int time) {
        int mod = 2 * (n - 1);
        int curr = time % mod;
        return 1 + (curr < n ? curr : mod-curr);
        
    }
};

File: Rust/solution2582.rs
struct Solution{}

impl Solution {
    pub fn pass_the_pillow(n: i32, time: i32) -> i32 {
        let m = 2 * (n - 1);
        let curr = time % m;

        if curr < n 
            {curr+1}
        else
            {m-curr+1}
    }
}